internal/clientconn/conn.go

```diff
@@ -140,7 +146,28 @@ func (c *conn) run(ctx context.Context) (err error) {
 	}()
 
 	bufr := bufio.NewReader(c.netConn)
+
+	// if test record path is set, split netConn reader to write to file and bufr
+	if c.testRecordPath != "" {
+
+		if err := os.MkdirAll(c.testRecordPath, 0o644); err != nil {
+			return err
+		}
+
+		path := filepath.Join(c.testRecordPath, "record.log")
+
+		//nolint:nosnakecase// os package require to use underscore
+		f, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o744)
+		if err != nil {
+			return err
+		}
+
+		r := io.TeeReader(c.netConn, f)
+		bufr = bufio.NewReader(r)
+	}
```

Comment:
@AlekSi  in place of `else` statement - do we want to redeclare bufr variable like here, or use switch statement?

Reply:
I think the current code is fine